lib/bundle/brew_installer.rb

```diff
@@ -1,18 +1,52 @@
 module Bundle
   class BrewInstaller
     def self.install(name, options = {})
+      new(name, options).install_or_upgrade
+    end
+
+    def initialize(name, options = {})
       if Bundle.brew_installed?
-        command = [name]
-        unless options[:args].nil?
-          options[:args].each do |arg|
-            command << "--#{arg}"
-          end
-        end
-
-        Bundle.system "brew", "install", *command
+        @name = name
+        @args = prepare_args(options[:args])
       else
         raise "Unable to install #{name}. Homebrew is not currently installed on your system"
       end
     end
+
+    def install_or_upgrade
+      if installed?
+        upgrade!
+      else
+        install!
+      end
+    end
+
+    private
+
+    def installed?
+      Bundle.installed_formulae.include?(@name)
+    end
+
+    def upgradable?
+      Bundle.outdated_formulae.include?(@name)
+    end
+
+    def install!
+      Bundle.system("brew", "install", @name, *@args)
+    end
+
+    def upgrade!
+      return Bundle.system("brew", "upgrade", @name) if !install_only? && upgradable?
```

Comment:
@xu-cheng fixed in https://github.com/Homebrew/homebrew-bundle/commit/92f444c4f930b1bb8f755d93adbacfeb38d434af